packages/core/contracts/financial-templates/optimistic-distributor/OptimisticDistributor.sol

```diff
@@ -0,0 +1,420 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../common/implementation/AncillaryData.sol";
+import "../../common/implementation/Lockable.sol";
+import "../../common/implementation/MultiCaller.sol";
+import "../../common/interfaces/AddressWhitelistInterface.sol";
+import "../../merkle-distributor/implementation/MerkleDistributor.sol";
+import "../../oracle/implementation/Constants.sol";
+import "../../oracle/interfaces/FinderInterface.sol";
+import "../../oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../oracle/interfaces/OptimisticOracleInterface.sol";
+import "../../oracle/interfaces/StoreInterface.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title  OptimisticDistributor contract.
+ * @notice Allows sponsors to distribute rewards through MerkleDistributor contract secured by UMA Optimistic Oracle.
+ */
+contract OptimisticDistributor is Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    /********************************************
+     *  OPTIMISTIC DISTRIBUTOR DATA STRUCTURES  *
+     ********************************************/
+
+    // Represents reward posted by a sponsor.
+    struct Reward {
+        address sponsor;
+        IERC20 rewardToken;
+        uint256 maximumRewardAmount;
+        uint256 earliestProposalTimestamp;
+        uint256 optimisticOracleProposerBond;
+        uint256 optimisticOracleLivenessTime;
+        bytes32 priceIdentifier;
+        bytes customAncillaryData;
+    }
+
+    // Represents proposed rewards distribution.
+    struct Proposal {
+        uint256 rewardIndex;
+        uint256 timestamp;
+        bytes32 merkleRoot;
+        string ipfsHash;
+    }
+
+    /********************************************
+     *      INTERNAL VARIABLES AND STORAGE      *
+     ********************************************/
```

Comment:
this can probs be removed. these variables are not "internal" per say. consider removing this block or changing it to more accurately describe the state variables.

Reply:
fixed in 3c2941ddcef57e9e71c3497709fff5cadedca7f7